### DISCUSSION TITLE
Backmapping Carbohydrates to all atom

### DISCUSSION THREAD

**Discussion by fgrunewald** [marrink-lab/polyply_1.0#353] [Q&A]
In #341 @cesar-lopez-bau asked if it is possible to backmap carbohydrates from Martini to all-atom using the new frame-based backmapping tool. 

With the current code (i.e. no frames) it is possible to backmap the general global conformation from Martini to all-atom by providing the virtual-site coordinates as meta-mol coordinates when generating the all-atom structure. In this case, the carbohydrate templates will be placed at the same center of geometry and then rotationally aligned. All needed files for Dextran as examples can be found here:

[backmap_DEX.tar.gz](https://github.com/marrink-lab/polyply_1.0/files/13725836/backmap_DEX.tar.gz)

The example contains an all-atom structure, which has been forward mapped to Martini and then backmapped again to all-atom. As far as I can tell from this example the chirality on the sugar links is correct and the conformation looks reasonable. However, with the frame based backmapping of @jan-stevens it might be more precise. The example folder contains a small script to automatically extract all the needed template filesl. Note that during energy minization relatively strong dihedral restraints should be applied. 

@jan-stevens from this example you should be able to figure out how to setup the reference frame and the backmap the structure.

**Comment by cesar-lopez-bau** [comment]
I got it working for Dextran, let me know how to contribute the topology library

**Reply by fgrunewald** [reply]
@cesar-lopez-bau good to hear! The best way would be to make a fork and place the files into the `data/{force_field_name}` directory. At the moment polyply does not have a CHARMm36 one so you should create one. Alternatively, you can also attach the files here and we'll incorporate them.
